[chore] packages/core typecheck red — test 파일 node:* import @types/node 미설정 (TS2591)

## 현상

`packages/core` 에서 `pnpm exec tsc --noEmit -p tsconfig.json` (test 포함 typecheck) 실행 시 TS2591 다수:

```
src/scene/orbit-line-segments.test.ts(1,16): error TS2591: Cannot find name 'node:fs'. ... add 'node' to the types field in your tsconfig.
src/scene/r-phase-allowlist.test.ts(1,16): error TS2591: Cannot find name 'node:fs'. ...
```

test 파일이 `node:fs`/`node:path`/`node:url` 을 import 하나 `tsconfig.json` 의 `types` 에 `node` 가 없어 인식 실패.

## 영향 (낮음)

- **빌드 게이트는 통과**: 실제 빌드는 `tsconfig.build.json`(test 제외) 사용 → `pnpm build` / CI 정상. 본 typecheck red 는 IDE/수동 `tsc --noEmit` 에서만 노출.
- develop 시점부터 존재하는 **pre-existing baseline** (#713 PR 무관 — PR #714 reviewer + qa 둘 다 분리 권고).

## 제안 fix (택1)

- `tsconfig.json` 의 `compilerOptions.types` 에 `"node"` 추가 (`@types/node` 이미 의존성에 있으면)
- 또는 test 전용 tsconfig 분리 (`tsconfig.test.json` 에 node types)

## 비-범위

- 기능 변경 없음 (타입 설정만). 빌드 게이트 무관.

근거: PR [#714](https://github.com/coseo12/astro-simulator/pull/714) (#713 클릭 선택) reviewer + qa 공통 분리 권고.

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -50,6 +50,7 @@
     "@babylonjs/core": "^8.0.0"
   },
   "devDependencies": {
+    "@types/node": "^25.6.0",
     "@vitest/coverage-v8": "^4.1.4",
     "typescript": "^6.0.0",
     "vitest": "^4.1.4"

--- a/packages/core/src/scene/r-phase-allowlist.test.ts
+++ b/packages/core/src/scene/r-phase-allowlist.test.ts
@@ -246,7 +246,9 @@ describe('#598 — browser-verify-378-focus.mjs FOCUS_BODIES SSoT 정합', () =>
     const source = fs.readFileSync(verifyScriptPath, 'utf-8');
     const match = source.match(/const\s+FOCUS_BODIES\s*=\s*\[([^\]]+)\]/);
     expect(match, 'FOCUS_BODIES 선언 패턴을 찾지 못함').toBeTruthy();
-    const focusBodies = match![1]
+    // capture group [1] 은 위 toBeTruthy 로 match 존재 보장 + 정규식에 캡처 그룹이 있어 항상 존재
+    // (noUncheckedIndexedAccess 로 string|undefined 추론되므로 non-null assertion).
+    const focusBodies = match![1]!
       .split(',')
       .map((s) => s.trim().replace(/^['"]|['"]$/g, ''))
       .filter((s) => s.length > 0);
@@ -267,7 +269,7 @@ describe('#619 — r1-ui-regression-guard.mjs targetIds SSoT 정합', () => {
   function extractArray(source: string, name: string): string[] {
     const match = source.match(new RegExp(`const\\s+${name}\\s*=\\s*\\[([^\\]]+)\\]`));
     expect(match, `${name} 선언 패턴을 찾지 못함`).toBeTruthy();
-    return match![1]
+    return match![1]!
       .split(',')
       .map((s) => s.trim().replace(/^['"]|['"]$/g, ''))
       .filter((s) => s.length > 0);
@@ -357,7 +359,7 @@ describe('#617 — showInShortcutBar 메타 SSoT 정합', () => {
     );
     const block = source.match(/const\s+FOCUS_BUTTONS\s*=\s*\[([\s\S]+?)\];/);
     expect(block, 'FOCUS_BUTTONS 선언을 찾지 못함').toBeTruthy();
-    const ids = [...block![1].matchAll(/id:\s*'([^']+)'/g)].map((m) => m[1]);
+    const ids = [...block![1]!.matchAll(/id:\s*'([^']+)'/g)].map((m) => m[1]);
     expect(ids).toEqual(shortcutBodies);
   });
 
@@ -371,7 +373,7 @@ describe('#617 — showInShortcutBar 메타 SSoT 정합', () => {
     const extract = (name: string) => {
       const m = source.match(new RegExp(`const\\s+${name}\\s*=\\s*\\[([^\\]]*)\\]`));
       expect(m, `${name} 선언을 찾지 못함`).toBeTruthy();
-      return m![1]
+      return m![1]!
         .split(',')
         .map((s) => s.trim().replace(/^['"]|['"]$/g, ''))
         .filter((s) => s.length > 0);

--- a/packages/core/src/scene/solar-system-scene.ts
+++ b/packages/core/src/scene/solar-system-scene.ts
@@ -76,10 +76,10 @@ const FLOATING_ORIGIN_THRESHOLD_METERS = AU;
 /**
  * P11-A #288 — dev-only assert gate.
  *
- * core 패키지는 `@types/node` 를 의존으로 두지 않으므로 `process` 가 types 에 없다.
- * runtime 에 `globalThis.process?.env?.NODE_ENV` 를 안전하게 읽고, Next.js webpack 이
- * 빌드 시점에 `NODE_ENV` 를 리터럴 치환 → prod bundle 에선 이 함수가 `false` 를 반환해
- * 하위 assert 블록이 DCE 된다.
+ * `process` 는 브라우저 번들 runtime 에 존재하지 않을 수 있다 (#715 부터 `@types/node` 로
+ * 타입은 있으나 — core 는 브라우저 타겟이라 런타임 부재 가능). 따라서 `globalThis.process?.env?.NODE_ENV`
+ * 를 안전하게 읽고, Next.js webpack 이 빌드 시점에 `NODE_ENV` 를 리터럴 치환 → prod bundle 에선
+ * 이 함수가 `false` 를 반환해 하위 assert 블록이 DCE 된다.
  */
 function floatingOriginAssertEnabled(): boolean {
   const g = globalThis as { process?: { env?: { NODE_ENV?: string } } };
@@ -1192,7 +1192,7 @@ export function createSolarSystemScene(
     }
 
     // P11-A #288 DoD β v2 — dev 빌드 assert: **focus body** local 좌표 절대값 ≤ 1e5 m (100 km).
-    // core 패키지는 `process` 타입이 없으므로 runtime guard 사용. Next.js webpack 이 `process.env.NODE_ENV`
+    // `process` 는 브라우저 번들 런타임에 없을 수 있어 runtime guard 사용. Next.js webpack 이 `process.env.NODE_ENV`
     // 를 빌드 타임 치환 → prod bundle 에서는 `'development' !== 'production'` 이 false 가 되어 DCE.
     // SSR / 테스트 환경은 `globalThis.process` 유/무 가드로 안전 접근.
     //

--- a/packages/core/tsconfig.json
+++ b/packages/core/tsconfig.json
@@ -2,6 +2,7 @@
   "compilerOptions": {
     "target": "ES2022",
     "lib": ["ES2022", "DOM", "DOM.Iterable"],
+    "types": ["node"],
     "module": "ESNext",
     "moduleResolution": "bundler",
     "strict": true,

